pkg/core/status.go

```diff
@@ -34,7 +34,14 @@ func InitializeReplicaStatuses(jobStatus *apiv1.JobStatus, rtype apiv1.ReplicaTy
 func UpdateJobReplicaStatuses(jobStatus *apiv1.JobStatus, rtype apiv1.ReplicaType, pod *corev1.Pod) {
 	switch pod.Status.Phase {
 	case corev1.PodRunning:
-		jobStatus.ReplicaStatuses[rtype].Active++
+		if pod.DeletionTimestamp != nil {
+			// when node is not ready, the pod will be in terminating state.
+			// Count deleted Pods as failures to account for orphan Pods that
+			// never have a chance to reach the Failed phase.
+			jobStatus.ReplicaStatuses[rtype].Failed++
+		} else {
+			jobStatus.ReplicaStatuses[rtype].Active++
+		}
```

Comment:
I don't think that the deleting pod is failed one. Please let me know if I'm missing any cases.

Reply:
Hi, @tenzen-y, 

@
I found some code in `pkg/util/k8sutil/k8sutil.go` 
`func IsPodActive(p *v1.Pod) bool {
	return v1.PodSucceeded != p.Status.Phase &&
		v1.PodFailed != p.Status.Phase &&
		p.DeletionTimestamp == nil
}`
, it seems that job active count should not contains the pod in `Terminating` state.
And in kubernetes jobs implementation, the `Terminating` pod doesn't count as active either.
In some cases, if the node becomes not ready, we are not certain if  the pod is RUNNING or not...

Reply:
Uhm. You have a point. When the pod is running and has a DeletionTimestamp field, we should mark Failed to the Job.

Thank you for clarifying!